pulser/simulation.py

```diff
@@ -74,6 +74,11 @@ def __init__(self, sequence, sampling_rate=1.0):
         self._build_basis_and_op_matrices()
         self._construct_hamiltonian()
 
+    @property
+    def hamiltonian(self):
+        '''Get the Hamiltonian QObjEvo created from the sequence'''
+        return self._hamiltonian
```

Comment:
Bear in mind that you will be giving direct access to the stored hamiltonian anyway, because it is a mutable object. If you want to protect the object stored in `self._hamiltonian`, you should return a copy instead (I believe `.copy()` works).